superset/dashboards/filters.py

```diff
@@ -88,6 +90,11 @@ def apply(self, query: Query, value: Any) -> Query:
                 )
             )
         )
+        roles_based_query = (
+            db.session.query(Dashboard.id)
+            .join(Dashboard.roles)
+            .filter(and_(Dashboard.published.is_(True), Role.id.in_(get_user_roles())),)
```

Comment:
I think `get_user_roles()` will return `Role` object instead of role ids.

```suggestion
            .filter(
                and_(
                    Dashboard.published.is_(True),
                    Role.id.in_([x.id for x in get_user_roles()]),
                ),
            )
```